src/Compilers/CSharp/Portable/Symbols/Metadata/PE/ExternalAnnotations.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Text;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE
+{
+    // PROTOTYPE(NullableReferenceTypes): external annotations should be removed or fully designed/productized
+    internal class ExtraAnnotations
+    {
+        // APIs that are useful to annotate:
+        //   1) don't accept null input
+        //   2) can return null
+        //   3) never return null
+        internal static readonly Dictionary<string, ImmutableArray<ImmutableArray<bool>>> Annotations =
+            new Dictionary<string, ImmutableArray<ImmutableArray<bool>>>
+            {
+                { "System.Boolean System.Boolean.Parse(System.String)", Parameters(skip, Nullable(false)) },
```

Comment:
>System.Boolean System.Boolean.Parse(System.String) [](start = 19, length = 50)

This format should be fine for tests, but as a product feature I think it should be more precise so that false positives and negatives are avoided. Perhaps, assembly qualified name format used in metadata is more appropriate rather than SymbolDisplay.

Reply:
This is intentionally designed to be suitable for prototyping and not as a product feature.

Reply:
I'll add a comment

---
In reply to: [179824782](https://github.com/dotnet/roslyn/pull/25928#discussion_r179824782) [](ancestors = 179824782)

Reply:
This thread is marked as resolved with response:

> I'll add a comment

But I do not see the comment.

---
In reply to: [179863122](https://github.com/dotnet/roslyn/pull/25928#discussion_r179863122) [](ancestors = 179863122,179824782)

Reply:
The comment is on line 13